apps/pi-drone-detector/src/frame_sources.py

```
@@ -0,0 +1,467 @@
+        # Try Picamera2 first (modern Pi OS)
+        try:
+            from picamera2 import Picamera2
+
+            self._picam2 = Picamera2()
+            config = self._picam2.create_preview_configuration(
+                main={"size": (self._width, self._height), "format": "RGB888"}
+            )
+            self._picam2.configure(config)
+            self._picam2.start()
+            self._using_libcamera = True
+            return True
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)